src/content/docs/en/guides/cms/apostrophecms.mdx

```diff
@@ -6,9 +6,505 @@ stub: true
+
+### Creating the [...slug.astro] component and fetching Apostrophe data
+
+Since Apostrophe is responsible for managing URLs to content, including creating new content and pages on the fly, you will only need one top-level Astro page component: the `[...slug].astro` route.
+
+Create a `[...slug].astro` file inside the `src/pages` folder and add the following content:
+
+```js title="src/pages/[...slug].astro"
+---
+import aposPageFetch from '@apostrophecms/apostrophe-astro/lib/aposPageFetch.js';
+import AposLayout from '@apostrophecms/apostrophe-astro/components/layouts/AposLayout.astro';
+import AposTemplate from '@apostrophecms/apostrophe-astro/components/AposTemplate.astro';
+
+const aposData = await aposPageFetch(Astro.request);
+const bodyClass = `myclass`;
+
+if (aposData.redirect) {
+  return Astro.redirect(aposData.url, aposData.status);
+}
+if (aposData.notFound) {
+  Astro.response.status = 404;
+}
+---
+<AposLayout title={aposData.page?.title} {aposData} {bodyClass}>
+    <Fragment slot="standardHead">
+      <meta name="description" content={aposData.page?.seoDescription} />
+      <meta name="viewport" content="width=device-width, initial-scale=1" />
+      <meta charset="UTF-8" />
+    </Fragment>
+    <AposTemplate {aposData} slot="main"/>
+</AposLayout>
+```
+
+The integration comes with an `aposPageFetch` method that should be used to automatically fetch the relevant data for the current URL. This is used to populate the `aposData` object that is passed to the template components.
+
+The `AposLayout` component supplied by the integration provides layout framework for the templates and other page HTML such as `<head>` markup. It is organized into multiple named slots including:
+  - `startBody`
+  - `startHead`
+  - `standardHead`
+  - `extraHead`
+  - `beforeMain`
+  - `main`
+  - `afterMain`
+  - `endBody`
+
+In the example, `meta` information is being added to the `standardHead` slot with a fragment. The `AposTemplate` component is being used to add the requested Astro template into the `main` slot of the layout, passing in the `aposData` object as a prop.
+
+## Making a blog with Astro and ApostropheCMS
+
+With the integration set up, you can now create a blog with Astro and ApostropheCMS.
+
+### Prerequisites
+
+1. **An ApostropheCMS project** - You can create a new project or use an existing one, this tutorial will only be showing how to create a blog piece and piece-page-type. You will have to integrate any other existing project code independently.
+2. **An Astro project integrated with ApostropheCMS** - To create a project from scratch, see [integrating with Astro](#integrating-with-astro) for instructions on how to set up the integration.
+
+### Creating a blog piece and piece-page-type
+
+To create your blog piece and piece-page-type for their display, navigate to the root of your ApostropheCMS project in your terminal. Use the CLI tool to create the new piece and piece-page-type with the following command:
```

Comment:
I'll try to clarify throughout. I wasn't sure that the CLI tool needed to be explicitly installed as part of this tutorial since it is a step that occurs during the setup of an environment for developing an ApostropheCMS project. I'm not sure how to clarify that your ApostropheCMS project and Astro projects are separate codebases other than what is already stated in the prerequisites section.